From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JustMangoT <38831897+JustMangoT@users.noreply.github.com>
Date: Thu, 17 Mar 2022 11:01:42 +0700
Subject: [PATCH] Rebranding


diff --git a/build.gradle.kts b/build.gradle.kts
index fe0223e68aa2ad0ec0db0130482e33ee09bf755d..65e931e3e81808c174f0de39e22d142344190120 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -18,7 +18,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":pearl-api")) // Pearl // Pufferfish // Paper // Purpur
+    implementation(project(":atelier-api")) // Atelier // Pearl // Pufferfish // Paper // Purpur
     // Pufferfish start
     implementation("io.papermc.paper:paper-mojangapi:1.18.2-R0.1-SNAPSHOT") {
         exclude("io.papermc.paper", "paper-api")
@@ -89,7 +89,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-Pearl-$implementationVersion", // Pufferfish // Purpur // Pearl
+            "Implementation-Version" to "git-Atelier-$implementationVersion", // Pufferfish // Purpur // Pearl // Atelier
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index 03c2c6ac2b7092192de0b471bb4fe36c4e286775..ef0160a9bb31774a2f9dad6bb591feef83956843 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -20,7 +20,7 @@ import java.util.stream.StreamSupport;
 public class PaperVersionFetcher implements VersionFetcher {
     private static final java.util.regex.Pattern VER_PATTERN = java.util.regex.Pattern.compile("^([0-9\\.]*)\\-.*R"); // R is an anchor, will always give '-R' at end
     // Purpur start
-    private static final String DOWNLOAD_PAGE = "https://github.com/Pearl-Project/Pearl/releases"; // Pearl
+    private static final String DOWNLOAD_PAGE = "https://github.com/AtelierMC/Atelier/releases"; // Atelier
     private static int distance = -2; public int distance() { return distance; }
     // Purpur end
     private static @Nullable String mcVer;
@@ -33,8 +33,8 @@ public class PaperVersionFetcher implements VersionFetcher {
     @Nonnull
     @Override
     public Component getVersionMessage(@Nonnull String serverVersion) {
-        String[] parts = serverVersion.substring("git-Pearl-".length()).split("[-\\s]"); // Pearl
-        final Component updateMessage = getUpdateStatusMessage("Pearl-Project/Pearl", "master", parts[0]); // Pearl
+        String[] parts = serverVersion.substring("git-Atelier-".length()).split("[-\\s]"); // Atelier
+        final Component updateMessage = getUpdateStatusMessage("AtelierMC/Atelier", "master", parts[0]); // Atelier
         final Component history = getHistory();
 
         return history != null ? Component.join(net.kyori.adventure.text.JoinConfiguration.separator(Component.newline()), history, updateMessage) : updateMessage; // Purpur
@@ -47,7 +47,7 @@ public class PaperVersionFetcher implements VersionFetcher {
                 String result = matcher.group();
                 mcVer = result.substring(0, result.length() - 2); // strip 'R' anchor and trailing '-'
             } else {
-                org.bukkit.Bukkit.getLogger().warning("Unable to match version to pattern! Report to Pearl!"); // Pearl
+                org.bukkit.Bukkit.getLogger().warning("Unable to match version to pattern! Report to Atelier!"); // Atelier
                 org.bukkit.Bukkit.getLogger().warning("Pattern: " + VER_PATTERN.toString());
                 org.bukkit.Bukkit.getLogger().warning("Version: " + org.bukkit.Bukkit.getBukkitVersion());
             }
diff --git a/src/main/java/com/destroystokyo/paper/console/PaperConsole.java b/src/main/java/com/destroystokyo/paper/console/PaperConsole.java
index 893389bb642390ce33bd0ecebb178d544f5759ec..f1aac5ffa9f1c2184c54a12fd0df76f107637c93 100644
--- a/src/main/java/com/destroystokyo/paper/console/PaperConsole.java
+++ b/src/main/java/com/destroystokyo/paper/console/PaperConsole.java
@@ -17,7 +17,7 @@ public final class PaperConsole extends SimpleTerminalConsole {
     @Override
     protected LineReader buildReader(LineReaderBuilder builder) {
         builder
-                .appName("Pearl") // Pearl
+                .appName("Atelier") // Atelier
                 .variable(LineReader.HISTORY_FILE, java.nio.file.Paths.get(".console_history"))
                 .completer(new ConsoleCommandCompleter(this.server))
                 .option(LineReader.Option.COMPLETE_IN_WORD, true);
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 23bff2e531e6cb7aeb4658e9038f2011a8869b64..bcc5aa69a03026f246a90eb2b6580f57d91c6257 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1723,7 +1723,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return org.purpurmc.purpur.PurpurConfig.serverModName; // Pearl - Pearl > // Purpur - Purpur > // Pufferfish - Pufferfish > // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return org.purpurmc.purpur.PurpurConfig.serverModName; // Atelier - Atelier > // Pearl - Pearl > // Purpur - Purpur > // Pufferfish - Pufferfish > // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index b1be7fe428ff9847abb604054140cdf6539cdaab..1fe2c8e7fdf8fe5f688d7dbddc17d1e6e69cfc32 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -248,7 +248,7 @@ import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
 
 public final class CraftServer implements Server {
-    private final String serverName = "Pearl"; // Paper // Pufferfish // Purpur // Pearl
+    private final String serverName = "Atelier"; // Paper // Pufferfish // Purpur // Pearl // Atelier
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index fc9877036bc3619b5a5754dd149b0097f7d067af..d9c7a26328c4cb2cee7469909a0827278493c7df 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/me.minhh2792.pearl/pearl-api/pom.properties"); // Pufferfish // Purpur // Pearl
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/tech.ateliermc.atelier/atelier-api/pom.properties"); // Pufferfish // Purpur // Pearl // Atelier
         Properties properties = new Properties();
 
         if (stream != null) {
diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index d0c4be748b5d667f41d626082cc878c879adc951..a04dcd7dd3cb50737ff6259cc2a9a7627f21e000 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -235,7 +235,7 @@ public class PurpurConfig {
         if (!TimingsManager.hiddenConfigs.contains("settings.seed")) TimingsManager.hiddenConfigs.add("settings.seed");
     }
 
-    public static String serverModName = "Pearl"; // Pearl
+    public static String serverModName = "Atelier"; // Atelier
     private static void serverModName() {
         serverModName = getString("settings.server-mod-name", serverModName);
     }
