From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JustMangoT <38831897+JustMangoT@users.noreply.github.com>
Date: Tue, 1 Mar 2022 20:10:45 +0700
Subject: [PATCH] Relocate server files


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
index cd03fed32807fd943fdeee8571c1648b8bd567d0..03326e3aed3973ef359d559a438418dcddb9ad37 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
@@ -55,7 +55,7 @@ public class PufferfishConfig {
 	}
 
 	public static void load() throws IOException {
-		File configFile = new File("pufferfish.yml");
+		File configFile = new File("config", "pufferfish.yml"); // AtelierSMP - Relocate files
 
 		if (configFile.exists()) {
 			try {
diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 1e0d261439255091a6f61485c0747231fbd5b1db..8246fb02f58a4e3e4100f1121351cf67080376ac 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -141,12 +141,12 @@ public class Main {
             org.spigotmc.SpigotConfig.disabledAdvancements = spigotConfiguration.getStringList("advancements.disabled"); // Paper - fix SPIGOT-5885, must be set early in init
             // Paper start - fix SPIGOT-5824
             File file;
-            File userCacheFile = new File("usercache.json");
+            File userCacheFile = new File("storage", "usercache.json"); // AtelierSMP - Relocate files
             if (optionset.has("universe")) {
                 file = (File) optionset.valueOf("universe"); // CraftBukkit
                 userCacheFile = new File(file, "usercache.json");
             } else {
-                file = new File(bukkitConfiguration.getString("settings.world-container", "."));
+                file = new File("worlds"); // AtelierSMP - Use 'worlds' folder by default
             }
             // Paper end - fix SPIGOT-5824
             YggdrasilAuthenticationService yggdrasilauthenticationservice = new com.destroystokyo.paper.profile.PaperAuthenticationService(Proxy.NO_PROXY); // Paper
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 111553443a7247db92b15819e9bbd717e4ace61b..e1d403b39b15451ccc73cc5dfefdcf87306aa9ef 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -122,10 +122,12 @@ import org.bukkit.event.player.PlayerRespawnEvent;
 
 public abstract class PlayerList {
 
-    public static final File USERBANLIST_FILE = new File("banned-players.json");
-    public static final File IPBANLIST_FILE = new File("banned-ips.json");
-    public static final File OPLIST_FILE = new File("ops.json");
-    public static final File WHITELIST_FILE = new File("whitelist.json");
+    // AtelierSMP start - Relocate files
+    public static final File USERBANLIST_FILE = new File("storage", "banned-players.json");
+    public static final File IPBANLIST_FILE = new File("storage", "banned-ips.json");
+    public static final File OPLIST_FILE = new File("storage", "ops.json");
+    public static final File WHITELIST_FILE = new File("storage", "whitelist.json");
+    // AtelierSMP end
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int SEND_PLAYER_INFO_INTERVAL = 600;
     private static final SimpleDateFormat BAN_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 782adba2a14d5b1a2e43c61909e776f208becf38..25a50d06143c2b9189af56740d6adbdec9bf62e8 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1073,7 +1073,10 @@ public final class CraftServer implements Server {
 
     @SuppressWarnings({ "unchecked", "finally" })
     private void loadCustomPermissions() {
-        File file = new File(this.configuration.getString("settings.permissions-file"));
+        // AtelierMC start - Relocate files
+        String path = this.configuration.getString("settings.permissions-file");
+        File file = new File((path != null && path.contains("/")) ? path : "config/" + path);
+        // AtelierMC end
         FileInputStream stream;
 
         try {
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 1726788b1364e5a16903c3ce7ad4559b8075495b..88bf1b0b94e1b0e5dfec8193c6dbf4d26b62c3dd 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -110,13 +110,13 @@ public class Main {
                 acceptsAll(Main.asList("b", "bukkit-settings"), "File for bukkit settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("bukkit.yml"))
+                        .defaultsTo(new File("config", "bukkit.yml")) // AtelierSMP - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(Main.asList("C", "commands-settings"), "File for command settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("commands.yml"))
+                        .defaultsTo(new File("config", "commands.yml")) // AtelierSMP - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(Main.asList("forceUpgrade"), "Whether to force a world upgrade");
@@ -135,7 +135,7 @@ public class Main {
                 acceptsAll(Main.asList("S", "spigot-settings"), "File for spigot settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("spigot.yml"))
+                        .defaultsTo(new File("config", "spigot.yml")) // AtelierSMP - Relocate files
                         .describedAs("Yml file");
                 // Spigot End
 
@@ -143,7 +143,7 @@ public class Main {
                 acceptsAll(asList("paper", "paper-settings"), "File for paper settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("paper.yml"))
+                        .defaultsTo(new File("config", "paper.yml")) // AtelierSMP - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(asList("add-plugin", "add-extra-plugin-jar"), "Specify paths to extra plugin jars to be loaded in addition to those in the plugins folder. This argument can be specified multiple times, once for each extra plugin jar path.")
@@ -157,7 +157,7 @@ public class Main {
                 acceptsAll(asList("purpur", "purpur-settings"), "File for purpur settings")
                     .withRequiredArg()
                     .ofType(File.class)
-                    .defaultsTo(new File("purpur.yml"))
+                    .defaultsTo(new File("config", "purpur.yml")) // AtelierSMP - Relocate files
                     .describedAs("Yml file");
                 // Purpur end
 
@@ -298,6 +298,7 @@ public class Main {
                 // Paper end
                 System.setProperty( "library.jansi.version", "Paper" ); // Paper - set meaningless jansi version to prevent git builds from crashing on Windows
                 System.out.println("Loading libraries, please wait...");
+                java.nio.file.Files.createDirectories(java.nio.file.Paths.get("storage")); // AtelierSMP - Relocate files
                 net.minecraft.server.Main.main(options);
             } catch (Throwable t) {
                 t.printStackTrace();
diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 74b49f4c1649c161c3c785be7708f5970989f7f3..8473ed353f243496fbb7de368c23c4bdbd69229d 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -25,7 +25,7 @@ public class HelpYamlReader {
     public HelpYamlReader(Server server) {
         this.server = server;
 
-        File helpYamlFile = new File("help.yml");
+        File helpYamlFile = new File("config", "help.yml"); // AtelierMC - Relocate files
         YamlConfiguration defaultConfig = YamlConfiguration.loadConfiguration(new InputStreamReader(getClass().getClassLoader().getResourceAsStream("configurations/help.yml"), Charsets.UTF_8));
 
         try {
